x/foundation/keeper/exec.go

```diff
@@ -0,0 +1,91 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/line/lbm-sdk/types"
+	sdkerrors "github.com/line/lbm-sdk/types/errors"
+
+	"github.com/line/lbm-sdk/x/foundation"
+)
+
+// ensureMsgAuthz checks that if a message requires signers that all of them are equal to the given account address of the operator.
+func ensureMsgAuthz(msgs []sdk.Msg, operator sdk.AccAddress) error {
+	for _, msg := range msgs {
+		for _, signer := range msg.GetSigners() {
+			if !operator.Equals(signer) {
+				return sdkerrors.ErrUnauthorized.Wrapf("signer of msg is not the operator: %s, %s", signer, operator)
+			}
+		}
+	}
+	return nil
+}
+
+func (k Keeper) Exec(ctx sdk.Context, proposalID uint64) error {
+	proposal, err := k.GetProposal(ctx, proposalID)
+	if err != nil {
+		return err
+	}
+
+	if proposal.Status != foundation.PROPOSAL_STATUS_SUBMITTED &&
+		proposal.Status != foundation.PROPOSAL_STATUS_CLOSED {
+		return sdkerrors.ErrInvalidRequest.Wrapf("not possible with proposal status: %s", proposal.Status.String())
+	}
+
+	if proposal.Status == foundation.PROPOSAL_STATUS_SUBMITTED {
+		if err := k.doTallyAndUpdate(ctx, proposal); err != nil {
+			return err
+		}
+	}
+
+	// Execute proposal payload.
+	if proposal.Status == foundation.PROPOSAL_STATUS_CLOSED &&
+		proposal.Result == foundation.PROPOSAL_RESULT_ACCEPTED &&
+		proposal.ExecutorResult != foundation.PROPOSAL_EXECUTOR_RESULT_SUCCESS {
+		logger := ctx.Logger().With("module", fmt.Sprintf("x/%s", foundation.ModuleName))
```

Comment:
This `logger` is only called when the error of `doExecuteMsgs` is occurred. So How about changing the location to inner of error of `doExecuteMsgs`.

Reply:
The logger is here for the logs to be left in the original context.